daemon/daemon_solaris.go

```diff
@@ -67,13 +118,198 @@ func (daemon *Daemon) getCgroupDriver() string {
 }
 
 func (daemon *Daemon) adaptContainerSettings(hostConfig *containertypes.HostConfig, adjustCPUShares bool) error {
+	if hostConfig.CPUShares < 0 {
+		logrus.Warnf("Changing requested CPUShares of %d to minimum allowed of %d", hostConfig.CPUShares, solarisMinCPUShares)
+		hostConfig.CPUShares = solarisMinCPUShares
+	} else if hostConfig.CPUShares > solarisMaxCPUShares {
+		logrus.Warnf("Changing requested CPUShares of %d to maximum allowed of %d", hostConfig.CPUShares, solarisMaxCPUShares)
+		hostConfig.CPUShares = solarisMaxCPUShares
+	}
+
+	if hostConfig.Memory > 0 && hostConfig.MemorySwap == 0 {
+		// By default, MemorySwap is set to twice the size of Memory.
+		hostConfig.MemorySwap = hostConfig.Memory * 2
+	}
+
+	if hostConfig.ShmSize != 0 {
+		hostConfig.ShmSize = container.DefaultSHMSize
+	}
+	if hostConfig.OomKillDisable == nil {
+		defaultOomKillDisable := false
+		hostConfig.OomKillDisable = &defaultOomKillDisable
+	}
+
 	return nil
 }
 
+// UsingSystemd returns true if cli option includes native.cgroupdriver=systemd
+func UsingSystemd(config *Config) bool {
+	return false
+}
+
 // verifyPlatformContainerSettings performs platform-specific validation of the
 // hostconfig and config structures.
 func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.HostConfig, config *containertypes.Config, update bool) ([]string, error) {
 	warnings := []string{}
+	sysInfo := sysinfo.New(true)
```

Comment:
Not sure about ignoring these values, I think it might be better to error and fail.